### PR TITLE
Update Tier and Status constraints with Filter Command

### PR DESCRIPTION
**Pull request by itsme-zeix**

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/FilterCommandParser.java [hunk 1]
```diff
@@ -148,11 +148,23 @@ private List<Predicate<Client>> collectPredicates(ArgumentMultimap argMultimap)
         if (argMultimap.getValue(PREFIX_TIER).isPresent()) {
             String substring = parseFieldForFilterCommand(() -> parseField(() -> argMultimap.getValue(
                     PREFIX_TIER).get(), errors));
+            if (substring.isEmpty()) {
+                System.out.println("HERE");
```
Comment (FionaQY) [inline src/main/java/seedu/address/logic/parser/FilterCommandParser.java]: leyew shun is rambling help

File: src/main/java/seedu/address/model/tier/Tier.java [hunk 2]
```diff
@@ -8,8 +8,7 @@
  * Guarantees: immutable; name is valid as declared in {@link #isValidTierName(String)}
  */
 public class Tier {
-
-    public static final String MESSAGE_CONSTRAINTS = "Tier can be set to 'gold', 'silver', 'bronze', 'reject' or 'na'. "
+    public static final String MESSAGE_CONSTRAINTS = "Tier must be set to Gold, Silver, Bronze, Reject, or NA. "
```
Comment (FionaQY) [inline src/main/java/seedu/address/model/tier/Tier.java:11]: I think we should standardize the quotes for Tier and Status.

File: src/main/java/seedu/address/logic/parser/FilterCommandParser.java [hunk 3]
```diff
@@ -148,11 +148,22 @@ private List<Predicate<Client>> collectPredicates(ArgumentMultimap argMultimap)
         if (argMultimap.getValue(PREFIX_TIER).isPresent()) {
             String substring = parseFieldForFilterCommand(() -> parseField(() -> argMultimap.getValue(
                     PREFIX_TIER).get(), errors));
+            if (substring.isEmpty()) {
+                errors.add("Tier has not been provided any value to filter by.\n"
+                        + "Please specify a status label to filter by, such as 'Gold', 'Silver', 'Bronze', 'Reject', "
+                        + "or 'NA'\n"
+                        + "To filter for clients without a visible tier label beside their name: filter s/ NA");
+            }
```
Comment (FionaQY) [inline src/main/java/seedu/address/logic/parser/FilterCommandParser.java:156]: I think its best to extract this out and put this as a string at the top instead.

**Review by FionaQY — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/parser/FilterCommandParser.java (hunk 1 above)
Comment (itsme-zeix) [reply src/main/java/seedu/address/logic/parser/FilterCommandParser.java]: HAHAHAHAHAHAHAHAHAHA THIS COMMENT

File: src/main/java/seedu/address/logic/parser/FilterCommandParser.java (hunk 3 above)
Comment (itsme-zeix) [reply src/main/java/seedu/address/logic/parser/FilterCommandParser.java:156]: Good point, let me implement this

File: src/main/java/seedu/address/model/tier/Tier.java (hunk 2 above)
Comment (itsme-zeix) [reply src/main/java/seedu/address/model/tier/Tier.java:11]: Nice catch!

**Comment by itsme-zeix** [comment]
```java
if (substring.isEmpty()) {
    errors.add("Tier has not been provided any value to filter by.\n"
            + "Please specify a status label to filter by, such as 'Gold', 'Silver', 'Bronze', 'Reject', "
            + "or 'NA'\n"
            + "To filter for clients without a visible tier label beside their name: filter s/ NA");
}
```
Should this be included in the UG or do you think it will make it too cluttered?

**Comment by ZShunRen** [comment]
> ```java
> public static final String TIER_NO_VALUE_PROVIDED_MESSAGE = "Tier has not been provided any value to filter by.\n"
>         + "Please specify a tier label to filter by, such as 'Gold', 'Silver', 'Bronze', 'Reject', or 'NA'.\n"
>         + "To filter for clients without a visible tier label beside their name, use: filter t/ NA";
> 
> public static final String STATUS_NO_VALUE_PROVIDED_MESSAGE = "Status has not been provided any value "
>         + "to filter by.\n"
>         + "Please specify a status label to filter by, such as 'Urgent', 'Non_Urgent', or 'NA'.\n"
>         + "To filter for clients without a visible status label beside their name, use: filter s/ NA";
> ```
> 
> Should this be included in the UG or do you guys think it will make it too cluttered?

Yes it should still be included in the UG, if not later all these kinds of front-facing details may startle users.

**Pull request merged**